src/pal/src/include/pal/stackstring.hpp

```diff
@@ -144,6 +161,24 @@ class StackString
         return;
     }
     
+    //count Should not include the terminating null
+    BOOL Append(const T * buffer, SIZE_T count)
```

Comment:
Actually, the more I look at the code where you use the Append, the more I feel like it would be better to not to introduce it at all. It seems that at all places, you can easily estimate the upper bound of the buffer for calling the OpenStringBuffer and keep the original code while writing into the buffer.
The reason I don't feel good about the Append is that it would cause reallocations, which is costy and the user of the Append might not realize it. Moreover, using upper bound estimate with OpenStringBuffers makes it the only point of failure w.r.t. memory allocation, so you could remove the checks at strcat_s and similar and use plain strcat etc, since the buffer would be guaranteed to be large enough.